server/test/services/export/PdfExporterTest.java

```diff
@@ -2,98 +2,98 @@
-        .putString(Path.create("applicant.formValue"), APPLICANT_VALUE);
-  }
-
-  @Before
-  public void createInMemoryWriter() {
-    this.inMemoryBytes = new ByteArrayOutputStream();
-    this.writer = new OutputStreamWriter(inMemoryBytes, StandardCharsets.UTF_8);
-  }
-
-  @Test
-  public void fillOneFormEntry() throws IOException {
-    // Check that the form is as expected.
-    File basePdf = new File("test/services/export/base.pdf");
-    assertThat(basePdf.canRead()).isTrue();
-    PDDocument doc = PDDocument.load(basePdf);
-    PDField formfield = doc.getDocumentCatalog().getAcroForm().getField("formfield");
-    assertThat(formfield).isNotNull();
-    assertThat(formfield.getValueAsString()).isEmpty();
-
-    // Create exporter and perform services.export.
-    ExporterFactory exporterFactory = instanceOf(ExporterFactory.class);
-    PdfExporter exporters = exporterFactory.pdfExporter(this.fakeProgramWithPdfExport);
-    exporters.export(fakeApplicant, writer);
-    writer.close();
-
-    // Load output document and check value.
-    PDDocument outputDocument = PDDocument.load(inMemoryBytes.toByteArray());
-    PDField outputField = outputDocument.getDocumentCatalog().getAcroForm().getField("formfield");
-    assertThat(outputField).isNotNull();
-    assertThat(outputField.getValueAsString()).isEqualTo(APPLICANT_VALUE);
-  }
+
+
+
+public class PdfExporterTest extends AbstractExporterTest {
+
+    @Test
+    public void validatePDFExport() throws IOException, DocumentException {
+        createFakeQuestions();
+        createFakeProgram();
+        createFakeApplications();
+
+        PdfExporter exporter = instanceOf(PdfExporter.class);
+
+        byte[] result = exporter.export(applicationOne);
+        PdfReader pdfReader = new PdfReader(result);
+
+        StringBuilder textFromPDF = new StringBuilder();
+        int pages = pdfReader.getNumberOfPages();
+        for (int i = 1; i < pages; i++) {
+            textFromPDF.append(PdfTextExtractor.getTextFromPage(pdfReader, i));
+        }
+        pdfReader.close();
+
+        assertThat(textFromPDF).isNotNull();
+        List<String> linesFromPDF = Splitter.on('\n').splitToList(textFromPDF.toString());
+
+        assertThat(textFromPDF).isNotNull();
+        String name = applicationOne.getApplicantData().getApplicantName() == null? "<Anonymous Applicant>" : applicationOne.getApplicantData().getApplicantName();
+        String programName = applicationOne.getProgram().getProgramDefinition().adminName();
+        assertThat(linesFromPDF.get(0)).isEqualTo(name + " " +  "("+applicationOne.id+")");
+        assertThat(linesFromPDF.get(1)).isEqualTo("Program Name : " + programName);
+
+        List<String> linesFromStaticString  = Splitter.on("\n").splitToList(applicationOneString);
+
+        for(int i=3;i<linesFromPDF.size();i++) {
+            assertThat(linesFromPDF.get(i)).isEqualTo(linesFromStaticString.get(i));
+        }
+
+    }
+    public static String applicationOneString =
```

Comment:
suggestion (blocking): this should be a constant
```suggestion
    public static final String APPLICATION_ONE_STRING =
```